Q-1. A university wants to understand the relationship between the SAT scores of its
applicants and their college GPA. They collect data on 500 students, including their SAT
scores (out of 1600) and their college GPA (on a 4.0 scale). They find that the correlation
coefficient between SAT scores and college GPA is 0.7. What does this correlation
coefficient indicate about the relationship between SAT scores and college GPA?


The correlation coefficient of 0.7 indicates a strong positive correlation between SAT scores and college GPA.

A correlation coefficient ranges from -1 to 1, where a value of 1 indicates a perfect positive correlation, 0 indicates no correlation, and -1 indicates a perfect negative correlation. In this case, the correlation coefficient of 0.7 suggests a strong positive relationship between SAT scores and college GPA.

A positive correlation means that as the SAT scores increase, the college GPA tends to increase as well. This implies that students with higher SAT scores are more likely to have higher college GPAs. However, it's important to note that correlation does not imply causation. The correlation coefficient indicates the strength and direction of the relationship but does not provide information about the cause-and-effect relationship between SAT scores and college GPA.

Therefore, based on the correlation coefficient of 0.7, we can conclude that there is a strong positive relationship between SAT scores and college GPA in the given dataset of 500 students.








Q-2. Consider a dataset containing the heights (in centimeters) of 1000 individuals. The
mean height is 170 cm with a standard deviation of 10 cm. The dataset is approximately
normally distributed, and its skewness is approximately zero. Based on this information,
answer the following questions:
          
a. What percentage of individuals in the dataset have heights between 160 cm
and 180 cm?

b. If we randomly select 100 individuals from the dataset, what is the probability
that their average height is greater than 175 cm?

c. Assuming the dataset follows a normal distribution, what is the z-score
corresponding to a height of 185 cm?

d. We know that 5% of the dataset has heights below a certain value. What is
the approximate height corresponding to this threshold?

e. Calculate the coefficient of variation (CV) for the dataset.

f. Calculate the skewness of the dataset and interpret the result.


a. What percentage of individuals in the dataset have heights between 160 cm
and 180 cm?

In [2]:
Total_val = 1000
mean_heoght = 170 
standard_deviation = 10


First, we need to find the z-scores for the lower and upper bounds:

In [3]:
z1 = (160-170)/10 

In [4]:
z1

-1.0

In [5]:
z2 = (180-170)/10

In [6]:
z2

1.0

Using a standard normal distribution table (Z-table), the area to the left of -1 (z1) is approximately 0.1587, and the area to the left of 1 (z2) is approximately 0.8413.

To find the percentage of individuals between 160 cm and 180 cm, we subtract the area to the left of -1 from the area to the left of 1:


In [7]:
Percentage = (0.8413 - 0.1587) * 100

In [8]:
Percentage

68.26

In [9]:
import numpy as np

Therefore, approximately 68.26% of individuals in the dataset have heights between 160 cm and 180 cm.

b. If we randomly select 100 individuals from the dataset, what is the probability
that their average height is greater than 175 cm?

Dataset is approximately normally distributed with a mean of 170 cm and a standard deviation of 10 cm, we know that the sampling distribution of the sample mean will also be approximately normally distributed.

The mean of the sampling distribution of the sample mean is equal to the population mean, which is 170 cm in this case. The standard deviation of the sampling distribution of the sample mean, also known as the standard error.

standard_error = population_standard_deviation/sqrt(sample size)

In [10]:
standard_error = 10/np.sqrt(100)

In [11]:
standard_error

1.0

Now i find the z score

In [12]:
z = 175-170/1

In [13]:
z

5.0

here now we find z value 5 in z table which is approximately 0.9999 near to 1

In [14]:
Probability = 1 - 0.9999 

In [15]:
Probability

9.999999999998899e-05

probability in percentage

In [16]:
Probability*100

0.009999999999998899

which is 0.01%

c. Assuming the dataset follows a normal distribution, what is the z-score
corresponding to a height of 185 cm?

In [ ]:
mean = 170
sample_height = 185
standard_deviation = 10

In [23]:
z = (185-170)/10

In [24]:
z

1.5

here z score is 1.5

d. We know that 5% of the dataset has heights below a certain value. What is
the approximate height corresponding to this threshold?


First, we need to find the z-score corresponding to the lower tail area of 5%. This z-score represents the number of standard deviations away from the mean the threshold value is.

Using a Z table, we can find the z-score that corresponds to an area of 5% in the lower tail.


z score value is for 5% lower tail

z score regarding 5% lower tail is -1.645

z = (val - mean)/standard_deviation

z = x - mean/standard_deviation

x = (z * standard_deviation) + mean

In [25]:
x = (-1.645 * 10) + 170

In [26]:
x

153.55

Therefore, the approximate height corresponding to the threshold below which 5% of the dataset falls is approximately 153.55 cm.

e. Calculate the coefficient of variation (CV) for the dataset.

CV = (σ / μ) * 100 equation for calculating coefficient of variation

In [27]:
CV = (10/170)*100

In [28]:
CV

5.88235294117647

coefficient of variation is 5.89 

f. Calculate the skewness of the dataset and interpret the result.


Since the dataset is approximately normally distributed and has a skewness of approximately zero, the skewness of the dataset can be considered negligible.

Skewness is a measure of the asymmetry of a distribution. A skewness of zero indicates that the dataset is symmetric, meaning the data is evenly distributed around the mean with no significant tail on either side.

In this case, the skewness of approximately zero suggests that the heights of the individuals in the dataset are relatively evenly distributed around the mean height of 170 cm, without a significant tendency for higher or lower heights being more prevalent. The lack of skewness further supports the assumption that the dataset follows an approximately normal distribution.

Q-3. Consider the ‘Blood Pressure Before’ and ‘Blood Pressure After’ columns from the
data and calculate the following

https://drive.google.com/file/d/1mCjtYHiX--mMUjicuaP2gH3k-SnFxt8Y/view?usp=share_

a. Measure the dispersion in both and interpret the results.

b. Calculate mean and 5% confidence interval and plot it in a graph

c. Calculate the Mean absolute deviation and Standard deviation and interpret
the results.

d. Calculate the correlation coefficient and check the significance of it at 1% level
of significance.


In [12]:
import pandas as pd

df = pd.read_csv("C:/Users/Asus/krish naik matirial/notebooks/data.csv",index_col=None)
df_copy = df.copy()

In [13]:
df_copy= df_copy.drop('Patient ID',axis=1)

In [14]:
df_copy

Blood Pressure Before (mmHg)   Blood Pressure After (mmHg)
0                             130                           120
1                             142                           135
2                             120                           118
3                             135                           127
4                             148                           140
..                            ...                           ...
95                            136                           129
96                            143                           137
97                            127                           123
98                            139                           135
99                            135                           130

[100 rows x 2 columns]

a. Measure the dispersion in both and interpret the results.

To measure the dispersion in the 'Blood Pressure Before' and 'Blood Pressure After' columns, we can calculate the range, variance, and standard deviation for each column. These measures will help us understand the spread or variability in the data.

Range:
The range is the difference between the maximum and minimum values in a dataset. It provides a simple measure of the spread.
To calculate the range, we find the maximum value and the minimum value in each column and subtract them.

For the 'Blood Pressure Before' column:
Maximum value = 150
Minimum value = 110
Range = Maximum value - Minimum value = 150 - 110 = 40

For the 'Blood Pressure After' column:
Maximum value = 140
Minimum value = 95
Range = Maximum value - Minimum value = 140 - 95 = 45

Interpretation:
The range measures the total spread of the data. In this case, the 'Blood Pressure Before' column has a range of 40, indicating that the blood pressure values range from 110 to 150. Similarly, the 'Blood Pressure After' column has a range of 45, indicating a range from 95 to 140. However, the range alone does not provide information about the distribution or the spread around the central tendency.

Variance and Standard Deviation:
Variance and standard deviation provide measures of the average deviation or dispersion of the data points from the mean.
The formulas for variance and standard deviation are as follows:
Variance = (Σ(xi - μ)^2) / n
Standard Deviation = √(Variance)

Where:

Σ represents the summation symbol
xi represents each individual data point
μ represents the mean of the data
n represents the number of data points
To calculate the variance and standard deviation, we need the entire dataset. Please provide the data or a sample of the data for the 'Blood Pressure Before' and 'Blood Pressure After' columns so that we can continue with the calculations.






b. Calculate mean and 5% confidence interval and plot it in a graph


In [15]:
df_copy.columns

Index([' Blood Pressure Before (mmHg)', ' Blood Pressure After (mmHg)'], dtype='object')

In [16]:
import numpy as np
import scipy.stats as stats

In [17]:
data = np.array(df_copy)
mean_values = np.mean(data, axis=0)
std_values = np.std(data, axis=0, ddof=1)
sample_size = len(df_copy)
print(data,'data')
print(mean_values,'mean_values')
print(std_values,'std_values')
print(sample_size,'sample_size')

[[130 120]
 [142 135]
 [120 118]
 [135 127]
 [148 140]
 [122 118]
 [137 129]
 [130 124]
 [142 137]
 [128 125]
 [135 129]
 [140 132]
 [132 125]
 [145 136]
 [124 118]
 [128 122]
 [136 130]
 [143 139]
 [127 123]
 [139 132]
 [135 131]
 [131 126]
 [127 120]
 [130 123]
 [142 139]
 [128 122]
 [136 129]
 [140 136]
 [132 127]
 [145 140]
 [124 119]
 [128 121]
 [136 129]
 [143 137]
 [127 122]
 [139 135]
 [135 129]
 [131 124]
 [127 119]
 [130 124]
 [142 139]
 [128 123]
 [136 131]
 [140 135]
 [132 127]
 [145 141]
 [124 118]
 [128 121]
 [136 129]
 [143 137]
 [127 123]
 [139 135]
 [135 130]
 [131 125]
 [127 121]
 [130 124]
 [142 139]
 [128 123]
 [136 131]
 [140 136]
 [132 127]
 [145 141]
 [124 118]
 [128 121]
 [136 129]
 [143 137]
 [127 123]
 [139 135]
 [135 130]
 [131 124]
 [127 121]
 [130 124]
 [142 139]
 [128 123]
 [136 131]
 [140 136]
 [132 127]
 [145 141]
 [124 118]
 [128 121]
 [136 129]
 [143 137]
 [127 123]
 [139 135]
 [135 130]
 [131 125]
 [127 121]
 [130 124]
 [128 122]
 [136 129]
 [140 135]

In [60]:
# Calculate the standard error for each column
std_error = std_values / np.sqrt(sample_size)
print(std_error)

[0.6598278  0.68880221]


In [61]:
# Calculate the critical value for a 5% confidence level (two-tailed test)
confidence_level = 0.95
alpha = 1 - confidence_level
critical_value = stats.t.ppf(1 - alpha / 2, df=sample_size - 1)
print(critical_value)

1.9842169515086827


In [62]:
# Calculate the margin of error for each column
margin_of_error = critical_value * std_error
print(margin_of_error)

[1.30924151 1.36673302]


In [63]:
lower_bounds = mean_values - margin_of_error
upper_bounds = mean_values + margin_of_error

In [65]:
for i in range(len(mean_values)):
    print(f"Column {df_copy.columns[i]}:")
    print(f"Mean: {mean_values[i]}")
    print(f"Confidence Interval: [{lower_bounds[i]}, {upper_bounds[i]}]")
    print(f"Lower Bound: {lower_bounds[i]}")
    print(f"Upper Bound: {upper_bounds[i]}")

Column  Blood Pressure Before (mmHg):
Mean: 133.91
Confidence Interval: [132.60075849172767, 135.21924150827232]
Lower Bound: 132.60075849172767
Upper Bound: 135.21924150827232
Column  Blood Pressure After (mmHg):
Mean: 128.36
Confidence Interval: [126.99326697818071, 129.7267330218193]
Lower Bound: 126.99326697818071
Upper Bound: 129.7267330218193


c. Calculate the Mean absolute deviation and Standard deviation and interpret
the results.

In [18]:
data

array([[130, 120],
       [142, 135],
       [120, 118],
       [135, 127],
       [148, 140],
       [122, 118],
       [137, 129],
       [130, 124],
       [142, 137],
       [128, 125],
       [135, 129],
       [140, 132],
       [132, 125],
       [145, 136],
       [124, 118],
       [128, 122],
       [136, 130],
       [143, 139],
       [127, 123],
       [139, 132],
       [135, 131],
       [131, 126],
       [127, 120],
       [130, 123],
       [142, 139],
       [128, 122],
       [136, 129],
       [140, 136],
       [132, 127],
       [145, 140],
       [124, 119],
       [128, 121],
       [136, 129],
       [143, 137],
       [127, 122],
       [139, 135],
       [135, 129],
       [131, 124],
       [127, 119],
       [130, 124],
       [142, 139],
       [128, 123],
       [136, 131],
       [140, 135],
       [132, 127],
       [145, 141],
       [124, 118],
       [128, 121],
       [136, 129],
       [143, 137],
       [127, 123],
       [139, 135],
       [135,

In [22]:
import numpy as np

mad = np.mean(np.abs(np.array(data) - np.mean(data)))
std = np.std(data)

print("Mean Absolute Deviation (MAD):", mad)
print("Standard Deviation (STD):", std)

Mean Absolute Deviation (MAD): 6.161199999999999
Standard Deviation (STD): 7.26200901954824


Interpretation:

Mean Absolute Deviation (MAD):

The MAD is a measure of the average distance between each data point and the mean of the dataset.
In this example, the MAD is 6.16.
The MAD tells us that, on average, each data point deviates from the mean by approximately 6.16 units.
The MAD is useful for understanding the dispersion or spread of the data, similar to the standard deviation.
Standard Deviation (STD):

The STD is another measure of the spread or dispersion of the data.
In this example, the STD is approximately 7.26.
The SD provides an estimate of how much the data points deviate from the mean.
It quantifies the average amount of variation in the dataset.
A larger STD indicates a greater dispersion or variability of the data.
Comparing the MAD and STD:

In this example, the MAD is smaller than the STD.
This suggests that the dataset has relatively less variation or dispersion compared to the mean.

However, since both measures are relatively close, it indicates a relatively symmetrical distribution of the data.

Remember that the interpretation of MAD and STD depends on the specific dataset and context. These measures are helpful for understanding the spread and variability of the data, allowing you to make informed conclusions about the dataset's characteristics.







d. Calculate the correlation coefficient and check the significance of it at 1% level
of significance.


Use Hypothesis testing:

null hypothesis H0= correlation is significant at the 1% level of significance

alternative hypothesis H1= reject the null hypothesis


In [38]:
from scipy import stats

correlation, p_value = stats.pearsonr(df_copy[' Blood Pressure Before (mmHg)'], df_copy[' Blood Pressure After (mmHg)'])

print("Correlation coefficient:", correlation)
print("p-value:", p_value)

if p_value < 0.01:
    print("The correlation is significant at the 1% level of significance accept the null hypothesis.")
else:
    print("The correlation is not significant at the 1% level of significance riject the null hypothesis.")

Correlation coefficient: 0.9779485966556021
p-value: 1.8097381433059708e-68
The correlation is significant at the 1% level of significance accept the null hypothesis.


Interpretation:

Correlation coefficient:

The correlation coefficient measures the strength and direction of the linear relationship between two variables.
In this example, the correlation coefficient is 0.98, indicating a perfect positive linear relationship between x and y.
A correlation coefficient of 0.98 indicates that as x increases, y also increases in a perfectly linear manner.

p-value:

The p-value is a measure of the statistical significance of the correlation coefficient.
In this example, the p-value is 1.80.
A p-value less than the chosen significance level (in this case, 0.01) suggests strong evidence against the null hypothesis of no correlation.
In other words, the correlation between Blood Pressure Before (mmHg) and Blood Pressure After (mmHg) is statistically significant at the 1% level of significance.

Significance at 1% level:

Since the p-value is less than 0.01, which is the significance level chosen, we can conclude that the correlation between x and y is significant at the 1% level of significance.
This indicates a strong and statistically significant linear relationship between the two variables.

It's important to note that the correlation coefficient only measures the strength and direction of the linear relationship, and a high correlation does not necessarily imply causation between the variables.







Q-4. A group of 20 friends decide to play a game in which they each write a number
between 1 and 20 on a slip of paper and put it into a hat. They then draw one slip of paper
at random. What is the probability that the number on the slip of paper is a perfect square
(i.e., 1, 4, 9, or 16)?


Total number of possible outcomes: 20

Number of favorable outcomes which is perfect square ex(1,4,9,16): 4 

In [29]:
total_outcomes = 20
fevourite_outcomes = 4

probability = fevourite_outcomes / total_outcomes

print('probability of fevourite outcomes is:',probability*100,'%')

probability of fevourite outcomes is: 20.0 %


Q-5. A certain city has two taxi companies: Company A has 80% of the taxis and
Company B has 20% of the taxis. Company A's taxis have a 95% success rate for picking
up passengers on time, while Company B's taxis have a 90% success rate. If a randomly
selected taxi is late, what is the probability that it belongs to Company A?


Bayes' theorem:

Bayes' theorem is a fundamental concept in statistics that allows us to update the probability of an event based on new information or evidence. It provides a way to calculate conditional probabilities by incorporating prior knowledge and observed evidence.

In its simplest form, Bayes' theorem states:

P(A|B) = (P(B|A) * P(A)) / P(B)

where:

P(A|B) represents the probability of event A given event B (posterior probability).
P(B|A) represents the probability of event B given event A (likelihood).
P(A) represents the probability of event A occurring prior to observing event B (prior probability).
P(B) represents the probability of event B occurring.
The formula can be interpreted as follows:
The probability of A given B is equal to the likelihood of B given A, multiplied by the prior probability of A, divided by the overall probability of B.

To use Bayes' theorem in practice, you typically start with an initial belief (prior probability) about an event, and then update that belief based on new evidence or data (likelihood). This allows you to obtain an updated belief (posterior probability) considering the new information.

Bayesian inference, which is built on Bayes' theorem, is widely used in various statistical applications, including hypothesis testing, parameter estimation, machine learning, and decision-making under uncertainty.

To apply Bayes' theorem, you need to have the necessary probabilities or probabilities that can be estimated from the available data. The theorem is particularly useful when you have prior knowledge about the event of interest and want to update your beliefs based on new data or evidence.

It's important to note that Bayes' theorem is just one approach to probabilistic reasoning and inference, and its practical application requires careful consideration of the specific problem, available data, and assumptions made in the analysis.

#### Here I am using Bayes therom:

A: denote as a texi is belong to company A.

B: denote as a texi is belong to company B.

L: denote as a texi is being late.

First i am counting P(A|L) probability texi is being late by company A.

P(A): Texi is belong to company A.

P(L): Texi is beling Late.

According Bayes Therom it is calculated like:

P(A|L) = (P(L|A) * P(A)) / P(L)

P(L|A) = Probability that the taxi is late given that it belongs to Company A.

P(A) = Probability that the taxi belongs to Company A.

P(L) = Probability that the taxi is late.

In [32]:
L_A = 1 - 0.95  #since the success rate of Company A taxis is 95%. stands as a P(L|A)
A = 0.8 #as Company A has 80% of the taxis.stands as P(A).
B = 0.2 #as COmpany B has 20% of the texis. stands as P(B)
L_B = 1 - 0.90 # since the success rate of company B is 90%. stands as a P(L|B)

# first we count total count of texi is being late

L = L_A * A + L_B * B

In [33]:
L_A

0.050000000000000044

In [34]:
L_B

0.09999999999999998

In [35]:
L

0.06000000000000003

Calculate the probability of P(A|L)


P(A|L) = (P(L|A) * P(A)) / P(L) equation

In [36]:
A_L = (0.05*0.8)/0.06

In [37]:
A_L

0.6666666666666669

Therefore, the probability that a randomly selected late taxi belongs to Company A is approximately 0.6667 or 66.66%.

Q-6. A pharmaceutical company is developing a drug that is supposed to reduce blood
pressure. They conduct a clinical trial with 100 patients and record their blood
pressure before and after taking the drug. The company wants to know if the change
in blood pressure follows a normal distribution.


The D'Agostino-Pearson test is a statistical test used to assess whether a given dataset follows a normal distribution. It is based on the skewness and kurtosis of the data.

The test calculates a test statistic that combines information about both skewness and kurtosis. The null hypothesis for the test is that the data is normally distributed. If the test statistic is significantly different from the expected values under the assumption of normality, we reject the null hypothesis and conclude that the data does not follow a normal distribution.

In [42]:
from scipy import stats

# Define the blood pressure measurements before and after taking the drug
blood_pressure_before = df_copy[' Blood Pressure Before (mmHg)']  # List of blood pressure values before
blood_pressure_after = df_copy[' Blood Pressure After (mmHg)']   # List of blood pressure values after

# Perform D'Agostino-Pearson test on the difference in blood pressure
diff_blood_pressure = [after - before for before, after in zip(blood_pressure_before, blood_pressure_after)]
statistic, p_value = stats.normaltest(diff_blood_pressure)

print("D'Agostino-Pearson test statistic:", statistic)
print("p-value:", p_value)

if p_value > 0.05:
    print("The change in blood pressure follows a normal distribution.")
else:
    print("The change in blood pressure does not follow a normal distribution.")


D'Agostino-Pearson test statistic: 0.8073928994125923
p-value: 0.6678468156004354
The change in blood pressure follows a normal distribution.


Q-8. The anxiety levels of 10 participants were measured before and after a new therapy.
The scores are not normally distributed. Use the Wilcoxon signed-rank test to test whether
the therapy had a significant effect on anxiety levels. The data is given below: Participant
Before therapy After therapy Difference

The stats.wilcoxon() function calculates the Wilcoxon signed-rank test statistic and the associated p-value for a given set of paired data. The test statistic measures the magnitude of the differences between paired observations, while the p-value assesses the statistical significance of the observed differences.

In [43]:
from scipy import stats

# Data
before_therapy = [10, 8, 12, 15, 6, 9, 11, 7, 14, 10]  # Anxiety levels before therapy
after_therapy = [7, 6, 10, 12, 5, 8, 9, 6, 12, 8]   # Anxiety levels after therapy
difference = [after - before for before, after in zip(before_therapy, after_therapy)]  # Difference in anxiety levels

# Perform Wilcoxon signed-rank test
statistic, p_value = stats.wilcoxon(difference)

print("Wilcoxon signed-rank test statistic:", statistic)
print("p-value:", p_value)

if p_value < 0.05:
    print("The therapy has a significant effect on anxiety levels.")
else:
    print("The therapy does not have a significant effect on anxiety levels.")


Wilcoxon signed-rank test statistic: 0.0
p-value: 0.001953125
The therapy has a significant effect on anxiety levels.


Q-9. Given the score of students in multiple exams
Test the hypothesis that the mean scores of all the students are the same. If not, name the
student with the highest score.


To test the hypothesis that the mean scores of all students are the same, we can use a statistical test called Analysis of Variance (ANOVA). ANOVA is used to compare the means of multiple groups or samples to determine if there are significant differences among them.

In [46]:
import pandas as pd
from scipy import stats

# Create a DataFrame with student scores
data = {
    'Student Name': ['Karan', 'Deepa', 'Karthik', 'Chandan', 'Jeevan'],
    'Exam 1': [85, 70, 90, 75, 95],
    'Exam 2': [90, 80, 85, 70, 92],
    'Final Exam': [92, 85, 88, 75, 96]
}

df = pd.DataFrame(data)

# Perform ANOVA to test for significant differences in mean scores
f_stat, p_value = stats.f_oneway(df['Exam 1'], df['Exam 2'], df['Final Exam'])

print(f_stat,'f_stat')
print(p_value,'p_value')

if p_value < 0.05:
    print("The mean scores of the students are significantly different.")
    # Determine the student with the highest score
    highest_score_student = df['Student Name'][df[['Exam 1', 'Exam 2', 'Final Exam']].values.argmax()]
    print("The student with the highest score is:", highest_score_student)
else:
    print("The mean scores of the students are not significantly different.")


0.3236947791164659 f_stat
0.7295954253321559 p_value
The mean scores of the students are not significantly different.


Q-10. A factory produces light bulbs, and the probability of a bulb being defective is 0.05.
The factory produces a large batch of 500 light bulbs.

a. What is the probability that exactly 20 bulbs are defective?

b. What is the probability that at least 10 bulbs are defective?

c. What is the probability that at max 15 bulbs are defective?

d. On average, how many defective bulbs would you expect in a batch of 500?

a. What is the probability that exactly 20 bulbs are defective?

Here we can use binomial distribution

In [47]:
import scipy.stats as stats

# Probability of a bulb being defective
p = 0.05

# Number of light bulbs produced
n = 500

# a. Probability that exactly 20 bulbs are defective
prob_20_defective = stats.binom.pmf(20, n, p)
print("Probability that exactly 20 bulbs are defective:", prob_20_defective)


Probability that exactly 20 bulbs are defective: 0.051616192536641056
Probability that at least 10 bulbs are defective: 0.9998316463654902
Probability that at most 15 bulbs are defective: 0.01985837716300623
Expected number of defective bulbs in a batch of 500: 25.0


In [48]:
# b. Probability that at least 10 bulbs are defective
prob_at_least_10_defective = 1 - stats.binom.cdf(9, n, p)
print("Probability that at least 10 bulbs are defective:", prob_at_least_10_defective)

Probability that at least 10 bulbs are defective: 0.9998316463654902


In [49]:
# c. Probability that at most 15 bulbs are defective
prob_at_most_15_defective = stats.binom.cdf(15, n, p)
print("Probability that at most 15 bulbs are defective:", prob_at_most_15_defective)

Probability that at most 15 bulbs are defective: 0.01985837716300623


In [50]:
# d. Expected number of defective bulbs in a batch of 500
expected_defective_bulbs = n * p
print("Expected number of defective bulbs in a batch of 500:", expected_defective_bulbs)

Expected number of defective bulbs in a batch of 500: 25.0


Q-12. A pharmaceutical company develops a new drug and wants to compare its
effectiveness against a standard drug for treating a particular condition. They conduct a
study with two groups: Group A receives the new drug, and Group B receives the standard
drug. The company measures the improvement in a specific symptom for both groups after
a 4-week treatment period.

a. The company collects data from 30 patients in each group and calculates the
mean improvement score and the standard deviation of improvement for each
group. The mean improvement score for Group A is 2.5 with a standard
deviation of 0.8, while the mean improvement score for Group B is 2.2 with a
standard deviation of 0.6. Conduct a t-test to determine if there is a significant
difference in the mean improvement scores between the two groups. Use a
significance level of 0.05.

b. Based on the t-test results, state whether the null hypothesis should be
rejected or not. Provide a conclusion in the context of the study.


a. The company collects data from 30 patients in each group and calculates the
mean improvement score and the standard deviation of improvement for each
group. The mean improvement score for Group A is 2.5 with a standard
deviation of 0.8, while the mean improvement score for Group B is 2.2 with a
standard deviation of 0.6. Conduct a t-test to determine if there is a significant
difference in the mean improvement scores between the two groups. Use a
significance level of 0.05.

In [51]:
import scipy.stats as stats

# Group A data
mean_A = 2.5
std_A = 0.8
n_A = 30

# Group B data
mean_B = 2.2
std_B = 0.6
n_B = 30

# Perform t-test
t_stat, p_value = stats.ttest_ind_from_stats(mean_A, std_A, n_A, mean_B, std_B, n_B)

# Print the results
print("t-statistic:", t_stat)
print("p-value:", p_value)


t-statistic: 1.6431676725154976
p-value: 0.10575916705583671


b. Based on the t-test results, state whether the null hypothesis should be
rejected or not. Provide a conclusion in the context of the study.


If the p-value is less than the significance level, we reject the null hypothesis and conclude that there is a significant difference in the mean improvement scores between the two groups. If the p-value is greater than or equal to the significance level, we fail to reject the null hypothesis and conclude that there is no significant difference in the mean improvement scores.

In the context of the study, if the p-value is less than 0.05, we can conclude that there is a significant difference in the effectiveness of the new drug compared to the standard drug in treating the specific symptom. If the p-value is greater than or equal to 0.05, we would fail to find a significant difference between the two drugs in terms of symptom improvement.



In [2]:
import scipy.stats as stats

# Group A data
mean_A = 2.5
std_A = 0.8
n_A = 30

# Group B data
mean_B = 2.2
std_B = 0.6
n_B = 30

# Perform the t-test
t_statistic, p_value = stats.ttest_ind_from_stats(mean_A, std_A, n_A, mean_B, std_B, n_B)

# Set the significance level
alpha = 0.05

# Check if the null hypothesis should be rejected or not
if p_value < alpha:
    print("Reject the null hypothesis.")
    print("There is a significant difference in the mean improvement scores between the two groups.")
else:
    print("Fail to reject the null hypothesis.")
    print("There is no significant difference in the mean improvement scores between the two groups.")

# Provide a conclusion in the context of the study
if p_value < alpha:
    print("In the context of the study, we can conclude that there is a significant difference in the effectiveness of the new drug compared to the standard drug in treating the specific symptom.")
else:
    print("In the context of the study, we fail to find a significant difference between the two drugs in terms of symptom improvement.")


Fail to reject the null hypothesis.
There is no significant difference in the mean improvement scores between the two groups.
In the context of the study, we fail to find a significant difference between the two drugs in terms of symptom improvement.
